src/integrations/prefect-redis/prefect_redis/records.py

```diff
@@ -0,0 +1,168 @@
+                PREFECT_RECORD_STORE_REDIS_DB setting if not provided
+            username: The username to use when connecting to the Redis server; defaults
+                to the value of the PREFECT_RECORD_STORE_REDIS_USERNAME setting if not
+                provided
+            password: The password to use when connecting to the Redis server; defaults
+                to the value of the PREFECT_RECORD_STORE_REDIS_PASSWORD setting if not
+                provided
+            ssl: Whether to use SSL when connecting to the Redis server; defaults to the
+                value of the PREFECT_RECORD_STORE_REDIS_SSL setting if not provided
+        """
+        self.host = host or PREFECT_RECORD_STORE_REDIS_HOST.value()
+        self.port = port or PREFECT_RECORD_STORE_REDIS_PORT.value()
+        self.db = db or PREFECT_RECORD_STORE_REDIS_DB.value()
+        self.username = username or PREFECT_RECORD_STORE_REDIS_USERNAME.value()
+        self.password = password or PREFECT_RECORD_STORE_REDIS_PASSWORD.value()
+        self.ssl = ssl or PREFECT_RECORD_STORE_REDIS_SSL.value()
+        self.client = Redis(
+            host=self.host,
+            port=self.port,
+            db=self.db,
+            username=self.username,
+            password=self.password,
+        )
+        self._locks = {}
+
+    @staticmethod
+    def _lock_name_for_key(key: str) -> str:
+        return f"lock:{key}"
+
+    def read(
+        self, key: str, holder: Optional[str] = None
+    ) -> Optional[TransactionRecord]:
+        holder = holder or self.generate_default_holder()
+
+        if self.is_locked(key) and not self.is_lock_holder(key, holder):
+            self.wait_for_lock(key)
+
+        serialized_result = self.client.get(name=key)
+        if serialized_result is None:
+            return None
+        assert isinstance(serialized_result, bytes)
+        return TransactionRecord(
+            key=key, result=BaseResult.model_validate_json(serialized_result)
+        )
+
+    def write(self, key: str, result: BaseResult, holder: Optional[str] = None) -> None:
+        if self.is_locked(key) and not self.is_lock_holder(key, holder):
+            raise ValueError(
+                f"Cannot write to transaction with key {key} because it is locked by another holder."
+            )
+        ex = None
+        if (
+            expiration := getattr(result, "expiration", None)
+        ) is not None and isinstance(expiration, pendulum.DateTime):
+            ex = math.ceil((expiration - pendulum.now()).total_seconds())
+        serialized_result = result.model_dump_json()
+        self.client.set(name=key, value=serialized_result, ex=ex)
+
+    def exists(self, key: str) -> bool:
+        return bool(self.client.exists(key))
+
+    def supports_isolation_level(self, isolation_level: IsolationLevel) -> bool:
+        return isolation_level in {
+            IsolationLevel.READ_COMMITTED,
+            IsolationLevel.SERIALIZABLE,
+        }
```

Comment:
is this intentionally checking a subset of future `IsolationLevel` members? or do we intend to just check that it is any valid `IsolationLevel`?
```suggestion
    def supports_isolation_level(self, isolation_level: IsolationLevel) -> bool:
        return isolation_level in IsolationLevel
```

Reply:
I don't know if new isolation levels will be added or what their expected behavior will be, so I'm being explicit here on purpose to avoid false positives in the future.